x-pack/plugins/apm/public/context/LocationContext.tsx

```diff
@@ -15,15 +15,9 @@ const initialLocation = {} as Location;
 
 const LocationContext = createContext(initialLocation);
 const LocationProvider: React.FC<Props> = ({ history, children }) => {
-  const [location, setLocation] = useState(history.location);
-
-  useEffect(() => {
-    const unlisten = history.listen(updatedLocation => {
-      setLocation(updatedLocation);
-    });
-
-    return unlisten;
-  }, []);
+  const location = {
+    ...history.location
+  };
```

Comment:
Why create a new `location` object on every render? Why not just pass `history.location` through?

Reply:
`history.location` is mutable. I figure it's easy to make a copy here and prevent any issues w/ some other consumer accidentally mutating the `history.location` object downstream.

Reply:
> history.location is mutable

Not the one coming from react router:
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/history.md#history-is-mutable

The problem with this is that it will always cause consumers of LocationContext to update when _any_ parent of LocationContext re-renders (even if `location` stays the same)

Reply:
Oh I just saw `<LocationProvider history={history}>` is already like that - yeah, I don't think we should do that. Instead we should just get `location` which `Router` will pass as a prop to child components (and which will be immutable)